ga4gh/datamodel/reads.py

```diff
@@ -122,29 +139,81 @@ class SimulatedReadGroupSet(AbstractReadGroupSet):
     def __init__(
             self, parentContainer, localId, randomSeed=1, numReadGroups=1,
             numAlignments=2):
-        super(SimulatedReadGroupSet, self).__init__(parentContainer, localId)
+        super(SimulatedReadGroupSet, self).__init__(
+            parentContainer, localId)
+        self._numAlignments = numAlignments
         for i in range(numReadGroups):
             localId = "rg{}".format(i)
             readGroup = SimulatedReadGroup(
                 self, localId, randomSeed + i, numAlignments)
             self.addReadGroup(readGroup)
 
+    def getNumAlignedReads(self):
+        return self._numAlignments
+
+    def getNumUnalignedReads(self):
+        return 0
+
+    def getPrograms(self):
+        return []
+
 
 class HtslibReadGroupSet(datamodel.PysamDatamodelMixin, AbstractReadGroupSet):
     """
     Class representing a logical collection ReadGroups.
     """
-    def __init__(self, parentContainer, localId, dataDir):
+    def __init__(self, parentContainer, localId, dataFile):
         super(HtslibReadGroupSet, self).__init__(parentContainer, localId)
-        self._dataDir = dataDir
-        self._setAccessTimes(dataDir)
-        self._scanDataFiles(dataDir, ["*.bam"])
+        self._samFilePath = dataFile
+        samFile = self.getFileHandle(self._samFilePath)
+        if 'RG' not in samFile.header or len(samFile.header['RG']) == 0:
+            self._defaultReadGroup = True
+            readGroup = HtslibReadGroup(self, 'default')
+            self.addReadGroup(readGroup)
+        else:
+            self._defaultReadGroup = False
+            for readGroupHeader in samFile.header['RG']:
+                readGroup = HtslibReadGroup(self, readGroupHeader['ID'])
+                self.addReadGroup(readGroup)
 
-    def _addDataFile(self, path):
-        filename = os.path.split(path)[1]
-        localId = os.path.splitext(filename)[0]
-        readGroup = HtslibReadGroup(self, localId, path)
-        self.addReadGroup(readGroup)
+    def openFile(self, dataFile):
+        return pysam.AlignmentFile(dataFile)
+
+    def getSamFilePath(self):
+        """
+        Returns the file path of the sam file
+        """
+        return self._samFilePath
+
+    def isDefaultReadGroup(self):
```

Comment:
I find this slightly confusing (it implies that the ReadGroupSet itself is a default read group). Maybe a more transparent flag would be ``hasReadGroupTags()``? This would make the client code easier to understand I think.